staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -413,26 +419,23 @@ func (c *AvailableConditionController) processNextWorkItem() bool {
 	return true
 }
 
-func (c *AvailableConditionController) enqueue(obj *apiregistrationv1.APIService) {
-	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
-	if err != nil {
-		klog.Errorf("Couldn't get key for object %#v: %v", obj, err)
-		return
-	}
-
-	c.queue.Add(key)
-}
-
 func (c *AvailableConditionController) addAPIService(obj interface{}) {
 	castObj := obj.(*apiregistrationv1.APIService)
 	klog.V(4).Infof("Adding %s", castObj.Name)
-	c.enqueue(castObj)
+	if castObj.Spec.Service != nil {
+		c.rebuildAPIServiceCache()
+	}
+	c.queue.Add(castObj.Name)
 }
 
-func (c *AvailableConditionController) updateAPIService(obj, _ interface{}) {
-	castObj := obj.(*apiregistrationv1.APIService)
-	klog.V(4).Infof("Updating %s", castObj.Name)
-	c.enqueue(castObj)
+func (c *AvailableConditionController) updateAPIService(oldObj, newObj interface{}) {
+	oldCastObj := oldObj.(*apiregistrationv1.APIService)
+	klog.V(4).Infof("Updating %s", oldCastObj.Name)
+	castObj := newObj.(*apiregistrationv1.APIService)
+	if (castObj != nil && castObj.Spec.Service != nil || oldCastObj != nil && oldCastObj.Spec.Service != nil) && !reflect.DeepEqual(castObj.Spec.Service, oldCastObj.Spec.Service) {
```

Comment:
@deads2k what panic were you thinking of? I'm not sure why the `castObj != nil` checks are needed

Reply:
I will modify this part of code once @deads2k clarifies.